src/creatures/npcs/spawns/spawn_npc_xml.cpp

```diff
@@ -0,0 +1,218 @@
+					centerPos.y + pugi::cast<uint16_t>(childNode.attribute("y").value()),
+					centerPos.z
+				), radius);
+				oldNpcList.push_front(npc);
+			}
+		}
+	}
+	return true;
+}
+
+bool SpawnsNpcOld::loadCustomSpawnXml(const std::string& customFileName)
+{
+	if (!isLoaded()) {
+		SPDLOG_ERROR("[SpawnsNpcOld::loadCustomSpawnXml] - "
+                     "Trying to load custom spawn xml before game startup, "
+                     "fileName: {}", customFileName);
+		return false;
+	}
+
+	pugi::xml_document doc;
+	pugi::xml_parse_result result = doc.load_file(customFileName.c_str());
+	if (!result) {
+		printXMLError("Error - SpawnsNpcOld::loadCustomSpawnXml", customFileName, result);
+		return false;
+	}
+
+	for (auto spawnNode : doc.child("npcs").children()) {
+		Position centerPos(
+			pugi::cast<uint16_t>(spawnNode.attribute("centerx").value()),
+			pugi::cast<uint16_t>(spawnNode.attribute("centery").value()),
+			pugi::cast<uint16_t>(spawnNode.attribute("centerz").value())
+			);
+
+		int32_t radius;
+		pugi::xml_attribute radiusAttribute = spawnNode.attribute("radius");
+		if (radiusAttribute) {
+			radius = pugi::cast<int32_t>(radiusAttribute.value());
+		} else {
+			radius = -1;
+		}
+
+		if (!spawnNode.first_child()) {
+			SPDLOG_WARN("Empty spawn at position: {} with radius: {}", centerPos.toString(), radius);
+			continue;
+		}
+
+		for (auto childNode : spawnNode.children()) {
+			if (strcasecmp(childNode.name(), "npc") == 0) {
+				pugi::xml_attribute nameAttribute = childNode.attribute("name");
+				if (!nameAttribute) {
+					continue;
+				}
+
+				NpcOld* npc = NpcOld::createNpc(nameAttribute.as_string());
+				if (!npc) {
+					continue;
+				}
+
+				pugi::xml_attribute directionAttribute = childNode.attribute("direction");
+				if (directionAttribute) {
+					npc->setDirection(static_cast<Direction>(pugi::cast<uint16_t>(directionAttribute.value())));
+				}
+
+				npc->setMasterPos(Position(
+					centerPos.x + pugi::cast<uint16_t>(childNode.attribute("x").value()),
+					centerPos.y + pugi::cast<uint16_t>(childNode.attribute("y").value()),
+					centerPos.z
+					), radius);
+
+				customNpcList.push_front(npc);
+			}
+		}
+
+		for (NpcOld* npc : customNpcList) {
+			g_game.placeCreature(npc, npc->getMasterPos(), false, true);
+		}
+	}
+
+	return true;
+}
+
+void SpawnsNpcOld::startup()
+{
+	if (!isLoaded() || isStarted()) {
+		return;
+	}
+
+	for (NpcOld* npcOld : oldNpcList) {
+		g_game.placeCreature(npcOld, npcOld->getMasterPos(), false, true);
+	}
+	oldNpcList.clear();
+
+	started = true;
+}
+
+void SpawnsNpcOld::clear()
+{
+	setLoaded(false);
+	setStarted(false);
+	filename.clear();
+}
+
+bool SpawnsNpcOld::isInZone(const Position& centerPos, int32_t radius, const Position& pos)
+{
+	if (radius == -1) {
+		return true;
+	}
+
+	return ((pos.getX() >= centerPos.getX() - radius) && (pos.getX() <= centerPos.getX() + radius) &&
+	        (pos.getY() >= centerPos.getY() - radius) && (pos.getY() <= centerPos.getY() + radius));
+}
```

Comment:
Correct me if I'm wrong, but both spawn classes are pretty much the same, I think you can have only 1 and pass the type as parameter, no? I don't think this spawn_npc_xml is needed at all